tests/components/vesync/test_init.py

```diff
@@ -67,37 +71,115 @@ async def test_async_setup_entry__no_devices(
-        assert setups_mock.call_args.args[1] == [Platform.FAN, Platform.SENSOR]
-        assert setup_mock.call_count == 0
-    assert manager.login.call_count == 1
-    assert hass.data[DOMAIN][VS_MANAGER] == manager
-    assert not hass.data[DOMAIN][VS_SWITCHES]
-    assert hass.data[DOMAIN][VS_FANS] == [fan]
-    assert not hass.data[DOMAIN][VS_LIGHTS]
-    assert hass.data[DOMAIN][VS_SENSORS] == [fan]
+        assert list(setups_mock.call_args.args[1]) == [
+            Platform.SWITCH,
+            Platform.FAN,
+            Platform.LIGHT,
+            Platform.SENSOR,
+        ]
+        assert register_mock.call_count == 1
+        assert register_mock.call_args.args[0] == DOMAIN
+        assert register_mock.call_args.args[1] == SERVICE_UPDATE_DEVS
+        assert callable(register_mock.call_args.args[2])
+
+    assert hass.data[DOMAIN][VS_MANAGER] == manager_devices
+    assert len(hass.data[DOMAIN][VS_FANS]) == 1
+    assert len(hass.data[DOMAIN][VS_LIGHTS]) == 2
+    assert len(hass.data[DOMAIN][VS_SENSORS]) == 2
+    assert len(hass.data[DOMAIN][VS_SWITCHES]) == 2
+
+
+async def test_asynch_setup_entry__loaded_state(
+    hass: HomeAssistant,
+    setup_platform,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test the resulting setup state is as expected."""
+
+    # fan devices
+    states = {}
+    identifier = "air-purifier"
+    entities = get_entities(hass, identifier)
+    assert len(entities) == 3
+    states[identifier] = get_states(hass, entities)
+
+    identifier = "asd_sdfKIHG7IJHGwJGJ7GJ_ag5h3G55"
+    entities = get_entities(hass, identifier)
+    assert len(entities) == 2
+    states[identifier] = get_states(hass, entities)
+
+    identifier = "400s-purifier"
+    entities = get_entities(hass, identifier)
+    assert len(entities) == 4
+    states[identifier] = get_states(hass, entities)
+
+    identifier = "600s-purifier"
+    entities = get_entities(hass, identifier)
+    assert len(entities) == 4
+    states[identifier] = get_states(hass, entities)
+
+    assert states == snapshot(name="fans")
+
+    # bulb devices
+    states = {}
+    identifier = "dimmable-bulb"
+    entities = get_entities(hass, identifier)
+    assert len(entities) == 1
+    states[identifier] = get_states(hass, entities)
+
+    identifier = "tunable-bulb"
+    entities = get_entities(hass, identifier)
+    assert len(entities) == 1
+    states[identifier] = get_states(hass, entities)
+
+    assert states == snapshot(name="bulbs")
+
+    # outlet devices
+    states = {}
+    identifier = "outlet"
+    entities = get_entities(hass, identifier)
+    assert len(entities) == 7
+    states[identifier] = get_states(hass, entities)
+
+    assert states == snapshot(name="outlets")
+
+    # switch devices
+    states = {}
+    identifier = "dimmable-switch"
+    entities = get_entities(hass, identifier)
+    assert len(entities) == 1
+    states[identifier] = get_states(hass, entities)
+
+    assert states == snapshot(name="switches")
```

Comment:
This should be tested inside `test_switch.py`

Reply:
moved